### ISSUE TITLE
Generate warnings when linking python bindings

### ISSUE THREAD

**Issue by yuki-koyama** [yuki-koyama/mathtoolbox#59]
https://travis-ci.com/yuki-koyama/mathtoolbox/jobs/262658713#L299-L300
```
Scanning dependencies of target pymathtoolbox
[ 40%] Building CXX object CMakeFiles/pymathtoolbox.dir/src/pybind/python-bindings.cpp.o
[ 43%] Linking CXX shared module pymathtoolbox.cpython-36m-darwin.so
ld: warning: direct access in function 'pybind11_init_pymathtoolbox(pybind11::module&)' from file '/tmp/lto.o' to global weak symbol 'mathtoolbox::optimization::BayesianOptimizer::EvaluatePoint(Eigen::Matrix<double, -1, 1, 0, -1, 1> const&) const' from file 'libmathtoolbox.a(bayesian-optimization.cpp.o)' means the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.
ld: warning: direct access in function 'char const* std::__1::__find_end<bool (*)(char, char), char const*, char const*>(char const*, char const*, char const*, char const*, bool (*)(char, char), std::__1::random_access_iterator_tag, std::__1::random_access_iterator_tag)' from file '/tmp/lto.o' to global weak symbol 'std::__1::char_traits<char>::eq(char, char)' from file 'libmathtoolbox.a(gaussian-process-regression.cpp.o)' means the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.
[ 43%] Built target pymathtoolbox
```

This issue should be fixed.

**Comment by yuki-koyama** [comment]
One of the warnings has been resolved but the other warning is still remaining.
https://travis-ci.com/yuki-koyama/mathtoolbox/jobs/263025309#L299

**Comment by yuki-koyama** [comment]
Resolved (#60)
